src/System.Management.Automation/engine/parser/TypeInferenceVisitor.cs

```diff
@@ -1241,7 +1241,26 @@ object ICustomAstVisitor.VisitDoUntilStatement(DoUntilStatementAst doUntilStatem
 
         object ICustomAstVisitor.VisitAssignmentStatement(AssignmentStatementAst assignmentStatementAst)
         {
-            return assignmentStatementAst.Left.Accept(this);
+            if (assignmentStatementAst.Left is AttributedExpressionAst attributedExpression)
+            {
+                if (attributedExpression is ConvertExpressionAst convertExpression)
+                {
+                    return new List<PSTypeName>() { new(convertExpression.Type.TypeName) };
+                }
+
+                ExpressionAst child = attributedExpression.Child;
+                while (child is AttributedExpressionAst attributeChild)
+                {
+                    if (attributeChild is ConvertExpressionAst convert)
+                    {
+                        return new List<PSTypeName>() { new(convert.Type.TypeName) };
+                    }
+
+                    child = attributeChild;
+                }
+            }
```

Comment:
Can we be a bit short?
```suggestion
            ExpressionAst child = assignmentStatementAst.Left;
            while (child is AttributedExpressionAst attributeChild)
            {
                if (attributeChild is ConvertExpressionAst convert)
                {
                    return new List<PSTypeName>() { new(convert.Type.TypeName) };
                }

                child = attributeChild.Child;
            }
```